src/models/msg.js

```diff
@@ -31,6 +31,7 @@ function Msg(attr) {
 	_.defaults(this, attr, {
 		from: "",
 		id: id++,
+		links: [],
 		previews: [],
```

Comment:
I have thought about that, but honestly it will look clunky at different levels, if anything because fetching callback will have to find which preview needs to be updated, and do it in place. I agree that having a separate array for ordering is not perfect, but I don't think that would be a better/simpler solution :/